File: FDP/T3_A8_PinedoBerumenClaraRuby/src/Ejercicio12_RFC.java
import java.util.Scanner;

public class Ejercicio12_RFC {

	public static void main(String[] args) {
		
		Scanner entrada = new Scanner(System.in);
		
		System.out.println("==Programa para obtener el RFC de una paersona==");
		System.out.println();
		
		
		System.out.println("Ingresa tu nombre"); 
		String nombre = entrada.nextLine();
		
		System.out.println("Ingresa tu apellido Paterno");
		String apellidoP = entrada.nextLine();
		
		System.out.println("Ingresa tu apellido Materno");
		String apellidoM = entrada.nextLine();
		
		System.out.println("Ingresa tu fecha de nacimiento en formato dd/mm/aaaa");
		String fecha = entrada.nextLine();
		
		String apellidoP1 = apellidoP.substring(0, 1).toLowerCase();
		apellidoM = apellidoM.substring(0,1);
		nombre = nombre.substring(0,1);
		String año = fecha.substring(8,10);
		String mes = fecha.substring(3,5);
		String dia = fecha.substring(0,2);
		
		
		String vocales = "";
		
        for (int i = 1; i < apellidoP.length(); i++) {
        	char ascii = apellidoP.charAt(i);

            if ((ascii == 'a') || (ascii == 'e') || (ascii == 'i') || (ascii == 'o') || (ascii== 'u')) {

            	vocales = vocales + ascii;
            	    
            }

        }//for
        
        String vocal2 = vocales.substring(0, 1);
		
		String rfc = apellidoP1+vocal2+apellidoM+nombre+año+mes+dia;
			System.out.println("Su RFC es: " + rfc.toUpperCase());
		
		

	}

}

File: FDP/T5_A6_PinedoBerumenClaraRuby/Ejercicio12_RFC.py
'''
Created on 4 dic. 2019

@author: Ruby
'''
print("Programa que obtiene el RFC de una persona\n")

nombre = input("Ingresa tu nombre: ")
apellidoPaterno = input("Ingresa tu apellido paterno: ").lower()
apellidoMaterno = input("Ingresa tu apellido materno: ")
fecha = input("Ingresa tu fecha de nacimiento en formato dd/mm/aaaa")

nombre = nombre[0:1]
apellidoPaterno1 = apellidoPaterno[0:1]
apellidoMaterno = apellidoMaterno[0:1]
anio = fecha[8:10]
mes = fecha[3:5]
dia = fecha [0:2]

vocales = ""

for i in apellidoPaterno: 
    if (i == "a" or i == "e" or i == "i" or i == "o" or i == "u"):
        vocales = vocales + i
    
primerVocal= vocales[0:1]   

#print(f"vocal1 {primerVocal}") 

rfc = (apellidoPaterno1+primerVocal+apellidoMaterno+nombre+anio+mes+dia).upper()

print(f"Su RFC es: {rfc}")





